qst/build.gradle

```diff
@@ -0,0 +1,12 @@
+dependencies {
+    annotationProcessor 'org.immutables:value:2.8.1'
```

Comment:
General note:  This is a user-facing API.  It needs to do extensive null checking to save users debugging effort and to save us support effort.  Clearly not a worry for the prototype, but it is a concern for production.